src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java

```diff
@@ -86,17 +86,14 @@ public final JmxProxy connectAny(
       Collection<String> hosts,
       int connectionTimeout) throws ReaperException {
 
-    if (hosts == null || hosts.isEmpty()) {
-      throw new ReaperException("no hosts given for connectAny");
-    }
+    Preconditions.checkArgument(null != hosts && !hosts.isEmpty(), "no hosts provided to connectAny");
+
     List<String> hostList = new ArrayList<>(hosts);
     Collections.shuffle(hostList);
-    Iterator<String> hostIterator = hostList.iterator();
 
     for (int i = 0; i < 2; i++) {
-      while (hostIterator.hasNext()) {
```

Comment:
this iterator was not being "reset" for the second loop, so the second loop never did anything…